src/webhook/mutation/pod_mutator/pod_mutator.go

```diff
@@ -96,6 +97,12 @@ func (webhook *podMutatorWebhook) isInjected(mutationRequest *dtwebhook.Mutation
 }
 
 func (webhook *podMutatorWebhook) handlePodMutation(mutationRequest *dtwebhook.MutationRequest) error {
+
+	if !kubeobjects.GetFieldBool(mutationRequest.BaseRequest.Pod.Annotations, dtwebhook.AnnotationOneAgentInject,
+		mutationRequest.BaseRequest.DynaKube.FeatureAutomaticInjection()) {
+		return nil
+	}
```

Comment:
this should not be here, you should check it in the affected mutators, so at:
https://github.com/Dynatrace/dynatrace-operator/blob/master/src/webhook/mutation/pod_mutator/dataingest_mutation/mutator.go#L29-L33
https://github.com/Dynatrace/dynatrace-operator/blob/master/src/webhook/mutation/pod_mutator/oneagent_mutation/mutator.go#L32-L34

Reply:
this is so that we don't start mixing things together (as it was previously before the refactor of the webhook)

Reply:
If I do it just in the mutators (what I had before), it doesn't work, because the stuff that happens after the mutators have been iterated seems to  depend on what the mutators do. The `initContainer` requires an `injectionConfig` volume to exist, which is missing if only the mutators are skipped. Also the `initContainer` and other stuff should not be added to the pod if the injection is disabled.

Reply:
AH! that totally makes sense. 🤦 

This is an actual bug, which needs fixing, because this would happen if the user disabled all mutations with the pod annotations.
So here is what I think should be done: We should only do anything with the pod IF at least 1 mutator was enabled.
The `handlePodMutation` should look something like: 
```
func (webhook *podMutatorWebhook) handlePodMutation(mutationRequest *dtwebhook.MutationRequest) error {
	...
	isMutated := false
	for _, mutator := range webhook.mutators {
		if !mutator.Enabled(mutationRequest.BaseRequest) {
			continue
		}
		if err := mutator.Mutate(mutationRequest); err != nil {
			return err
		}
		isMutated = true
	}
	if !isMutated {
		log.Info("no mutators where enabled")
		return nil
	}
    ...
}
```

Reply:
fixed in
https://github.com/Dynatrace/dynatrace-operator/pull/914/commits/b9ff99b4c70a7bc436d2f72b088973b07adfb428